examples/threading/threading.c

```diff
@@ -28,6 +62,39 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      *
      * See implementation details in threading.h file comment block
      */
-    return false;
+     
+     // malloc structure
+     struct thread_data *t_data = (struct thread_data *)malloc(sizeof(struct thread_data));
+     
+     // check for availability of memory
+     if(t_data == NULL)
+     {
+     	ERROR_LOG("\nRunning out of memory\n");
+     	return false;
+     }
+     else
+     	DEBUG_LOG("\nMalloc successful\n");
+     
+     // assign structure variables with respective parameters/arguments of function
+     t_data->wait_to_obtain_ms.tv_sec = 0;
+     t_data->wait_to_obtain_ms.tv_nsec = (wait_to_obtain_ms*1000000);
+     t_data->wait_to_release_ms.tv_sec = 0;
+     t_data->wait_to_release_ms.tv_nsec = (wait_to_obtain_ms*1000000);
+     t_data->mutex = mutex;
+     
+     // return code
+     int rc;
+     
+     // creates a thread
+     rc = pthread_create(&t_data->thread, NULL, threadfunc, t_data);
+     if(rc == 0)
+     {
+     	DEBUG_LOG("\nCreate successful\n");
+     }
+     else
+     	ERROR_LOG("\nPthread create failed\n");
```

Comment:
@abaa3023 I think this logic is missing the requirement to return false if the thread could not be started.